cpp/src/slice2cpp/Gen.cpp

```diff
@@ -905,6 +920,25 @@ Slice::Gen::validateMetadata(const UnitPtr& u)
     };
     knownMetadata.emplace("cpp:ice_print", std::move(icePrintInfo));
 
+    // "cpp:identifier"
+    MetadataInfo identifierInfo = {
+        .validOn =
+            {typeid(InterfaceDecl),
+             typeid(Operation),
+             typeid(ClassDecl),
+             typeid(Slice::Exception),
+             typeid(Struct),
+             typeid(Sequence),
+             typeid(Dictionary),
+             typeid(Enum),
+             typeid(Enumerator),
+             typeid(Const),
+             typeid(Parameter),
+             typeid(DataMember)},
+        .acceptedArgumentKind = MetadataArgumentKind::SingleArgument,
+    };
+    knownMetadata.emplace("cpp:identifier", std::move(identifierInfo));
+
```

Comment:
This is the entry which allows the new `cpp:identifier` through the metadata validator.
Note that as-of-now, we do not support `cpp:identifier` on Modules.
But it is supported on every other relevant Slice construct.